definition/definition.go

```diff
@@ -31,14 +31,43 @@ type Chain interface {
 // carry on, call Chain.Continue() and pass the context.
 type Middleware func(context.Context, Chain) error
 
+// Operator is used to operate an object and return an replacement object.
+//
+// For example:
+//  A converter:
+//    type ConverterForAnObject struct{}
+//    func (c *ConverterForAnObject) Kind() {return "converter"}
+//    func (c *ConverterForAnObject) In() reflect.Type {return definition.TypeOf(&ObjectV1{})}
+//    func (c *ConverterForAnObject) Out() reflect.Type {return definition.TypeOf(&ObjectV2{})}
+//    func (c *ConverterForAnObject) Operate(ctx context.Context, object interface{}) (interface{}, error) {
+//        objV2, err := convertObjectV1ToObjectV2(object.(*ObjectV1))
+//        return objV2, err
+//    }
+//
+//  A validator:
+//    type ValidatorForAnObject struct{}
+//    func (c *ValidatorForAnObject) Kind() {return "validator"}
+//    func (c *ValidatorForAnObject) In() reflect.Type {return definition.TypeOf(&Object{})}
+//    func (c *ValidatorForAnObject) Out() reflect.Type {return definition.TypeOf(&Object{})}
+//    func (c *ValidatorForAnObject) Operate(ctx context.Context, object interface{}) (interface{}, error) {
+//        if err := validate(object.(*Object)); err != nil {
+//            return nil, err
+//        }
+//        return object, nil
+//    }
 type Operator interface {
-	Operate(ctx context.Context, object interface{}) (interface{}, error)
-}
-
-type OperatorFunc func(ctx context.Context, object interface{}) (interface{}, error)
-
-func (f OperatorFunc) Operate(ctx context.Context, object interface{}) (interface{}, error) {
-	return f(ctx, object)
+	// Kind indicates operator type.
+	Kind() string
+	// In returns the type of the only object parameter of operator.
+	// The type must be a concrete struct or built-in type. It should
+	// not be interface unless it's a file or stream reader.
+	In() reflect.Type
+	// Out returns the type of the only object result of operator.
+	// The type must be a concrete struct or built-in type. It should
+	// not be interface unless it's a file or stream reader.
+	Out() reflect.Type
+	// Operate operates an object and return one.
+	Operate(ctx context.Context, field string, object interface{}) (interface{}, error)
```

Comment:
What is field?

Reply:
Field is the field name for operators to generate error. It may be `app`, `X-Tenant` and so on.